numba/typed/typedobjectutils.py

```diff
@@ -197,3 +197,78 @@ def build_wrapper(fn):
     builder.ret(context.get_constant(types.int32, -1))
 
     return equal_fn
+
+
+def _get_primitive_equal(context, module, datamodel, element_type):
+    assert not datamodel.contains_nrt_meminfo()
+
+    data_ptr_ty = ir.IntType(8).as_pointer()
+    equal_fnty = ir.FunctionType(ir.IntType(32), [data_ptr_ty, data_ptr_ty])
+    equal_fn = cgutils.get_or_insert_function(
+        module,
+        equal_fnty,
+        name='.numba_{}.{}_equal'.format(
+            context.fndesc.mangled_name, element_type),
+    )
+    builder = ir.IRBuilder(equal_fn.append_basic_block())
```

Comment:
You're creating a new builder here, but you can just re-use the one from `_dict_set_method_table` codegen.

Reply:
Emm, I tried naively reusing the builder from `_dict_set_method_table`, the replace this `ir.IRBuilder(..)` with `builder.append_basic_block()`, then I got a lowering error.
Looks like I didn't point this builder to the first bb in the new function. Am I missing anything?

BTW, the creation way is also from the legacy code in this file.